json_journal/data_analysis.py

```diff
@@ -82,3 +85,74 @@ def total_profit(full_account_name_balances):
             pass
     total = (income * (-1)) - expenses
     return total
+
+
+def calculate_profit(ledger):
+    '''
+    Calculate the profit of a ledger file.
+
+    Parameters
+    ----------
+    Ledger : list of dicts
+        Loaded json file.
+
+    Returns
+    -------
+    Profit_calculated : int
+        Value of total profit.
+    '''
+    profit_sorted_ledger = sorted(ledger, key=lambda d: d['date'])
+    for transaction in profit_sorted_ledger:
+        entries = transaction['entries']
+        profit_calculated = 0
```

Comment:
I am worried about the scope of this variable. It is defined in the loop, but returned? Is it being re-defined every iteration?

Reply:
done - it should have been outside the loop